modules/core/src/main/java/org/apache/ignite/internal/causality/VersionedValue.java

```diff
@@ -276,8 +276,8 @@ public void set(long causalityToken, T value) {
             history.put(causalityToken, initFut);
         }
 
-        assert actualToken0 == NOT_INITIALIZED || actualToken0 + 1 == causalityToken : IgniteStringFormatter.format(
-                "Token must be greater than actual by exactly 1 [token={}, actual={}]", causalityToken, actualToken0);
+        assert actualToken0 == NOT_INITIALIZED || actualToken0 <= causalityToken : IgniteStringFormatter.format(
```

Comment:
Here and bellow. Seems that according to current logic of assignments key registration on start it's not possible to have `actualToken0 == causalityToken`, only `actualToken0 < causalityToken` is valid.
Also `Token must be greater than or equal to actual ->  Token must be greater than`
WDYT?

Reply:
Fixed